sdk/core/azure-core-amqp/src/amqp/link.cpp

```diff
@@ -466,11 +466,11 @@ namespace Azure { namespace Core { namespace Amqp { namespace _detail {
     LinkImpl* link = static_cast<LinkImpl*>(context);
     if (link->m_eventHandler)
     {
-#if defined(BUILD_TESTING)
```

Comment:
@LarryOsterman , can you help me with this CI error. I wouldn't expect the BUILD_TESTING CMake flag to be used here, directly:
https://dev.azure.com/azure-sdk/public/_build/results?buildId=3669258&view=logs&j=aa5e5880-f49c-5532-bf1f-326a5a437391&t=d994038f-2a9a-51dc-9535-ccadb48b8b7e&l=203
```text
[ 19%] Building CXX object sdk/attestation/azure-security-attestation/CMakeFiles/azure-security-attestation.dir/src/attestation_client_options.cpp.o
/mnt/vss/_work/1/s/sdk/core/azure-core-amqp/src/amqp/link.cpp: In static member function ‘static void Azure::Core::Amqp::_detail::LinkImpl::OnLinkFlowOnFn(void*)’:
/mnt/vss/_work/1/s/sdk/core/azure-core-amqp/src/amqp/link.cpp:470:42: error: cannot convert ‘Azure::Core::Amqp::_detail::Link’ to ‘const std::shared_ptr<Azure::Core::Amqp::_detail::LinkImpl>&’
  470 |       link->m_eventHandler->OnLinkFlowOn(Link{link->shared_from_this()});
      |                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                          |
      |                                          Azure::Core::Amqp::_detail::Link
In file included from /mnt/vss/_work/1/s/sdk/core/azure-core-amqp/src/amqp/link.cpp:12:
/mnt/vss/_work/1/s/sdk/core/azure-core-amqp/src/amqp/private/link_impl.hpp:35:64: note:   initializing argument 1 of ‘virtual void Azure::Core::Amqp::_detail::LinkImplEvents::OnLinkFlowOn(const std::shared_ptr<Azure::Core::Amqp::_detail::LinkImpl>&)’
   35 |     virtual void OnLinkFlowOn(std::shared_ptr<LinkImpl> const& link) = 0;
      |                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~

```

Reply:
Before your changes, the BUILD_TESTING macro was never defined when building AMQP. Because the *BUILD_TESTING macro was private to the package, the AMQP library used the BUILD_TESTING_AMQP macro. 

After your changes, the _azure_BUILD_TESTING macro is defined when building AMQP.

Before your changes, the existence of the BUILD_TESTING at line 469 was a mistake but it didn't make a difference because BUILD_TESTING was never defined when building AMQP.

After your changes, you enabled the code at line 470, which could never compile.

Reply:
I'll also admit to being surprised that you managed to get this to compile locally before you ran it in the CI Pipeline.

There's basically no way this code could ever compile.

Reply:
> There's basically no way this code could ever compile.

That's what I figured. I filed an issue on this and will revert this line of change as it was a clean-up I noticed and picked up, can you please take a look separately, to see what the intended compiler fix is.
https://github.com/Azure/azure-sdk-for-cpp/issues/5495